packages/realm/src/Realm.ts

```diff
@@ -1080,50 +1175,325 @@ function isEmbedded(objectSchema: binding.ObjectSchema): boolean {
 type TypesType = typeof Types;
-type UserType = typeof User;
-type CredentialsType = typeof Credentials;
-type ConfigurationType = Configuration;
-type FlexibleSyncConfigurationType = FlexibleSyncConfiguration;
+type UpdateModeType = UpdateMode;
+type UserType = User;
+type UserStateType = UserState;
+
+type BaseSubscriptionSetType = BaseSubscriptionSet;
+type LogLevelType = LogLevel;
+type NumericLogLevelType = NumericLogLevel;
+type PartitionValueType = PartitionValue;
+type SubscriptionOptionsType = SubscriptionOptions;
+type SubscriptionSetType = SubscriptionSet;
+type SubscriptionsStateType = SubscriptionsState;
+type SubscriptionType = Subscription;
+type SyncSessionType = SyncSession;
+
+type ObjectIdType = BSON.ObjectId;
+type Decimal128Type = BSON.Decimal128;
+type UUIDType = BSON.UUID;
+
+type ApiKeyType = ApiKey;
+type EmailPasswordAuthType = EmailPasswordAuth;
+type ApiKeyAuthType = ApiKeyAuth;
+
+type GlobalDate = Date;
 
 // eslint-disable-next-line @typescript-eslint/no-namespace
-export namespace Realm {
+export declare namespace Realm {
+  // TODO: Decide if we want to deprecate this as well
   export type Object<T = DefaultObject> = RealmObject<T>;
-  export type Collection<T = unknown> = CollectionType<T>;
-  export type OrderedCollection<T = unknown> = OrderedCollectionType<T>;
-  export type Results<T = unknown> = ResultsType<T>;
-  export type List<T = unknown> = ListType<T>;
-  export type Dictionary<T = unknown> = DictionaryType<T>;
-  export type Set<T = unknown> = SetType<T>;
-  export type App = AppType;
-  export type UpdateMode = UpdateModeType;
-  export type ObjectSchema = ObjectSchemaType;
-  /**
-   * @deprecated Will be removed in v13.0.0. Please use {@link PropertySchema}.
-   */
-  export type ObjectSchemaProperty = ObjectSchemaPropertyType;
-  export type PropertySchema = PropertySchemaType;
-  export type PropertySchemaShorthand = PropertySchemaShorthandType;
-  export type Mixed = unknown;
-  export type BSON = BSONType;
-  export type Types = TypesType;
-  export type User = UserType;
-  export type Credentials = CredentialsType;
-  export type Configuration = ConfigurationType;
-  export type FlexibleSyncConfiguration = FlexibleSyncConfigurationType;
+  export {
+    // Pure type exports below
+    /** @deprecated Please use named imports */
+    AppType as App,
+    /** @deprecated Please use named imports */
+    AppChangeCallback,
+    /** @deprecated Please use named imports */
+    AppConfiguration,
+    /** @deprecated Please use named imports */
+    BaseConfiguration,
+    /** @deprecated Please use named imports */
+    BaseObjectSchema,
+    /** @deprecated Please use named imports */
+    BaseSyncConfiguration,
+    /** @deprecated Please use named imports */
+    BSONType as BSON,
+    /** @deprecated Please use named imports */
+    CanonicalObjectSchemaProperty,
+    /** @deprecated Please use named imports */
+    CanonicalPropertiesTypes,
+    /** @deprecated Please use named imports */
```

Comment:
> for methods that return canonical types (e.g. Realm.schema) we could change those to return the relaxed ones.

From my work on Studio I found it nicer to program against the canonical types. Granted, most end-user applications doesn't do schema reflection like Studio does, but for a CMS like front-end to a database, having the canonical types as well as having the `Realm#schema` be canonical is actually a massive upside, since the consuming code needs to consider less edge-cases.

Reply:
Interesting, point well taken! :) (ps. I believe `CanonicalObjectSchema` needs to be added there as well then)